tools/tests/com.liferay.ide.portlet.ui.tests/src/com/liferay/ide/portlet/ui/tests/LiferayPortletWizardTests.java

```diff
@@ -0,0 +1,1303 @@
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import com.liferay.ide.core.util.FileUtil;
+import com.liferay.ide.portlet.ui.tests.page.ContainerSelectionPageObject;
+import com.liferay.ide.portlet.ui.tests.page.CreateLiferayPortletWizardPageObject;
+import com.liferay.ide.portlet.ui.tests.page.CssSelectionPageObject;
+import com.liferay.ide.portlet.ui.tests.page.IconSelectionPageObject;
+import com.liferay.ide.portlet.ui.tests.page.InterfaceSelectionPageObject;
+import com.liferay.ide.portlet.ui.tests.page.JavaScriptSelectionPageObject;
+import com.liferay.ide.portlet.ui.tests.page.LiferayPortletDeploymentDescriptorPageObject;
+import com.liferay.ide.portlet.ui.tests.page.ModifiersInterfacesMethodStubsPageObject;
+import com.liferay.ide.portlet.ui.tests.page.NewSourceFolderPageObject;
+import com.liferay.ide.portlet.ui.tests.page.PackageSelectionPageObject;
+import com.liferay.ide.portlet.ui.tests.page.PortletDeploymentDescriptorPageObject;
+import com.liferay.ide.portlet.ui.tests.page.SelectTypePageObject;
+import com.liferay.ide.portlet.ui.tests.page.SuperclassSelectionPageObject;
+import com.liferay.ide.project.ui.tests.NewProjectPageObject;
+import com.liferay.ide.project.ui.tests.ProjectWizardTests;
+import com.liferay.ide.project.ui.tests.page.CreateProjectWizardPageObject;
+import com.liferay.ide.sdk.core.SDK;
+import com.liferay.ide.sdk.core.SDKManager;
+import com.liferay.ide.sdk.core.SDKUtil;
+import com.liferay.ide.ui.tests.ProjectWizard;
+import com.liferay.ide.ui.tests.SWTBotBase;
+import com.liferay.ide.ui.tests.UITestsUtils;
+import com.liferay.ide.ui.tests.swtbot.page.DialogPageObject;
+import com.liferay.ide.ui.tests.swtbot.page.EditorPageObject;
+import com.liferay.ide.ui.tests.swtbot.page.TextEditorPageObject;
+import com.liferay.ide.ui.tests.swtbot.page.TreeItemPageObject;
+
+import java.io.File;
+import java.util.Arrays;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.swtbot.eclipse.finder.SWTWorkbenchBot;
+import org.eclipse.swtbot.swt.finder.keyboard.Keyboard;
+import org.eclipse.swtbot.swt.finder.keyboard.KeyboardFactory;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Ashley Yuan
+ */
+public class LiferayPortletWizardTests extends SWTBotBase implements LiferayPortletWizard, ProjectWizard
+{
+
+    protected static IPath getCustomLocationBase()
+    {
+        final IPath customLocationBase =
+            org.eclipse.core.internal.utils.FileUtil.canonicalPath( new Path( System.getProperty( "java.io.tmpdir" ) ) ).append(
+                "custom-project-location-tests" );
+
+        return customLocationBase;
+    }
+
+    @BeforeClass
+    public static void setupPluginsSDK() throws Exception

```

Comment:
This method seems to have a lot of duplicated code with others.  Setting up the plugins SDK should be a shared superclass method where the specific parts are used as configuration or abstract methods.